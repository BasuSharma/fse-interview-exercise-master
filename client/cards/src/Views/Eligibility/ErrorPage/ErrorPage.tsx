import styled from "styled-components";
import View from "../../../DesignSystem/View";

const ResultsWrapper = styled.div`
  flex: 1 1 auto;
  padding-top: 48px;
  justify-content: center;
  margin: 0 -8px;
  display: flex;
  flex-wrap: wrap;
`;

const HeadingText = styled.div`
  flex: 1 1 auto;
  width: 80%;
  padding-top: 10px;
  padding-bottom:10px;
  justify-content: center;
  display: flex;
  flex-wrap: wrap;
  border-radius: 1px;
  border: solid .5px #eee;
  min-height: 50px;
  font-size: 25px;
  font-weight: bold;
  text-align: center;
  color: rgb(42, 98, 231, 0.75);
`;

// Error  page to show generic handled errors related to application 
const ErrorPage = () => {
  
  
     return <ResultsWrapper>
        <View>
        <HeadingText>Error Occured. please try agian later or contact support. </HeadingText>
         </View>
       </ResultsWrapper>;
   

 
};
  
export default ErrorPage;


