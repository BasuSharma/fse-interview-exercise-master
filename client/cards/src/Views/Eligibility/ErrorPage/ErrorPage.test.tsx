import React from 'react';
import { screen } from '@testing-library/react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ErrorPage from './ErrorPage';

Enzyme.configure({ adapter: new Adapter() })


it("renders correctly", () => {
  const tree = shallow(<ErrorPage />);
  expect(toJson(tree)).toMatchSnapshot();
});
