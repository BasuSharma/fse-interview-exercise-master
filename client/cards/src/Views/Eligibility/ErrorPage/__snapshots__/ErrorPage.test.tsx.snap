// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Error Occured. please try agian later or contact support. 
    </styled.div>
  </styled.div>
</styled.div>
`;
